### PR TITLE
Inbox.Actor and String Like refactoring

### PR DESCRIPTION
**Pull request by nbparrell**
These are the changes asked for by the ticket https://github.com/akkadotnet/akka.net/issues/265

Made all public methods private and also rewrote the Receive so that it is using is with if else instead of a switch.

### CODE REVIEW

File: src/core/Akka/Actor/Inbox.Actor.cs [hunk 1]
```diff
@@ -105,49 +105,57 @@ public bool MessagePredicate(object message)
         /// <returns>TBD</returns>
         protected override bool Receive(object message)
         {
-            switch (message)
-            {
-                case Get get:
+            /*var getMessage = message is Get;
+            var selectMessage = message is Select;
+            var startWatch = message is StartWatch;
+            var stopWatch = message is StopWatch;
```
Comment (marcpiechura) [inline src/core/Akka/Actor/Inbox.Actor.cs]: Please remove comments

File: src/core/Akka/Actor/Inbox.Actor.cs [hunk 2]
```diff
@@ -105,49 +105,57 @@ public bool MessagePredicate(object message)
...
+            var kick = message is Kick;*/
+            //var 
+            //Console.WriteLine("Entered the switch");
+            if(message != null){
```
Comment (marcpiechura) [inline src/core/Akka/Actor/Inbox.Actor.cs]: No need for null check, „is“ handles it automatically.

File: src/core/Akka/Actor/Inbox.Actor.cs [hunk 3]
```diff
@@ -105,49 +105,57 @@ public bool MessagePredicate(object message)
...
+                if(message is Get){
                     if (_messages.Count == 0)
                     {
-                        EnqueueQuery(get);
+                        EnqueueQuery((Get)message);
```
Comment (marcpiechura) [inline src/core/Akka/Actor/Inbox.Actor.cs]: Please use the new syntax and define a variable name for the casted get directly in the if statement, I.e. „ if ( message is Get get)“

File: src/core/Akka/Actor/Inbox.Actor.cs (hunk 2 above)
Comment (marcpiechura) [reply src/core/Akka/Actor/Inbox.Actor.cs]: And please don’t use java style for the curlybraces, put them in the next line.

File: src/core/Akka/Actor/Inbox.Actor.cs [hunk 4]
```diff
@@ -5,69 +5,69 @@
...
 {
     /// <summary>
     /// TBD
     /// </summary>
-    internal class InboxActor : ActorBase
+    internal class InboxActor:ActorBase 
     {
-        private readonly InboxQueue<object> _messages = new InboxQueue<object>();
-        private readonly InboxQueue<IQuery> _clients = new InboxQueue<IQuery>();
+        private readonly InboxQueue < object > _messages = new InboxQueue < object > ();
```
Comment (marcpiechura) [inline src/core/Akka/Actor/Inbox.Actor.cs]: Something is wrong with your auto formatting rules, we don't want to have a space in the generic type parameter ;)

File: src/core/Akka/Util/StringLike.cs [hunk 5]
```diff
@@ -26,7 +25,15 @@ public static bool Like(this string text,string pattern, bool caseSensitive = fa
...
-            pattern = pattern.Replace("*", ".*?");
+            if(pattern[0] == '*' && pattern[pattern.Length-1]!='*'){
+                pattern = pattern.Replace("*", "(.*?)");
+                pattern = pattern + "(?!.)";
+            }else if(pattern[0]!='*' && pattern[pattern.Length-1]=='*'){
+                pattern = pattern.Replace("*", "(.*?)");
+                pattern = "^"+pattern;
+            }else{
+                pattern = pattern.Replace("*", "(.*?)");
+            }
```
Comment (marcpiechura) [inline src/core/Akka/Util/StringLike.cs]: Could you please move this into its own PR ? Makes its easier to reason about the changes.

Comment (nbparrell) [reply src/core/Akka/Util/StringLike.cs]: Not sure what you mean by PR? Not completely up on the short hand.

File: src/core/Akka/Actor/Inbox.Actor.cs (hunk 4 above)
Comment (nbparrell) [reply src/core/Akka/Actor/Inbox.Actor.cs]: Will do.

File: src/core/Akka/Util/StringLike.cs (hunk 5 above)
Comment (marcpiechura) [reply src/core/Akka/Util/StringLike.cs]: Soory, PR means Pull Request. In the end it would be good to have a pull request per issue it handles.

**Comment by marcpiechura** [comment]
@nbparrell just wanted to know if you’re still working on this one.

**Comment by nbparrell** [comment]
@marcpiechura Yeah, I will be finishing this, as the summer started things got kinda hectic. Will make changes asap.

**Comment by Danthar** [comment]
@nbparrell minor poke ;)

**Comment by Aaronontheweb** [comment]
Replaced via #4073
